pyteal/ast/maybe.py

```diff
@@ -29,62 +30,23 @@ def __init__(
             immediate_args (optional): Immediate arguments for the op. Defaults to None.
             args (optional): Stack arguments for the op. Defaults to None.
         """
-        super().__init__()
-        self.op = op
-        self.type = type
-        self.immediate_args = immediate_args if immediate_args is not None else []
-        self.args = args if args is not None else []
-        self.slotOk = ScratchSlot()
-        self.slotValue = ScratchSlot()
+        types = [TealType.uint64, type]
+        super().__init__(op, types, immediate_args=immediate_args, args=args)
 
     def hasValue(self) -> ScratchLoad:
         """Check if the value exists.
 
         This will return 1 if the value exists, otherwise 0.
         """
-        return self.slotOk.load(TealType.uint64)
+        return self.output_slots[0].load(self.types[0])
```

Comment:
If you wanted to be super backwards compatible, you could use the `@property` decorator to create getter functions for `slotOk` and `slotValue` that return the appropriate value from `self.output_slots`.

It shouldn't be strictly necessary since those fields were not meant to be used by consumers (though I have written some hacky code that does access the slots directly), but there's no downside to it.